File: server/services/templateService.js
class TemplateService {
  constructor() {
    this.templates = {
      'standard-cim': {
        id: 'standard-cim',
        name: 'Standard CIM Template',
        description: 'Comprehensive CIM template for investment banking',
        sections: [
          'executiveSummary',
          'companyOverview',
          'marketAnalysis',
          'financialAnalysis',
          'investmentHighlights',
          'roiProjections',
          'riskFactors',
          'appendices'
        ]
      },
      'tech-startup': {
        id: 'tech-startup',
        name: 'Technology Startup CIM',
        description: 'Specialized template for technology companies',
        sections: [
          'executiveSummary',
          'technologyOverview',
          'marketOpportunity',
          'businessModel',
          'financialProjections',
          'competitiveAnalysis',
          'teamAndAdvisors',
          'riskFactors'
        ]
      },
      'manufacturing': {
        id: 'manufacturing',
        name: 'Manufacturing CIM Template',
        description: 'Template optimized for manufacturing companies',
        sections: [
          'executiveSummary',
          'operationsOverview',
          'marketPosition',
          'financialPerformance',
          'facilitiesAndCapacity',
          'supplyChain',
          'growthStrategy',
          'riskFactors'
        ]
      }
    };
  }

  async getTemplate(templateId) {
    const template = this.templates[templateId];
    if (!template) {
      throw new Error(`Template not found: ${templateId}`);
    }
    return template;
  }

  async getAllTemplates() {
    return Object.values(this.templates);
  }

  async validateTemplate(templateId) {
    return !!this.templates[templateId];
  }
}

module.exports = new TemplateService();

File: backend/services/template_service.py
"""
Template Service - CIM template management.
Migrated from server/services/templateService.js.
"""
from typing import Dict, Any, List, Optional
from utils.logger import logger


class TemplateService:
    """Service for managing CIM templates."""
    
    def __init__(self):
        """Initialize available templates."""
        self.templates = {
            "standard": {
                "id": "standard",
                "name": "Standard CIM",
                "description": "Complete CIM with all standard sections",
                "sections": [
                    "executive_summary",
                    "company_overview",
                    "financial_analysis",
                    "market_analysis",
                    "investment_highlights",
                    "risk_factors",
                    "roi_projections"
                ]
            },
            "quick": {
                "id": "quick",
                "name": "Quick Overview",
                "description": "Condensed CIM for preliminary review",
                "sections": [
                    "executive_summary",
                    "financial_analysis",
                    "investment_highlights"
                ]
            },
            "detailed": {
                "id": "detailed",
                "name": "Detailed Analysis",
                "description": "In-depth CIM with extended financial modeling",
                "sections": [
                    "executive_summary",
                    "company_overview",
                    "financial_analysis",
                    "market_analysis",
                    "competitive_analysis",
                    "investment_highlights",
                    "risk_factors",
                    "roi_projections",
                    "management_team",
                    "appendix"
                ]
            }
        }
    
    def get_template(self, template_id: str) -> Optional[Dict[str, Any]]:
        """Get template by ID."""
        return self.templates.get(template_id)
    
    def get_all_templates(self) -> List[Dict[str, Any]]:
        """Get all available templates."""
        return list(self.templates.values())
    
    def get_template_sections(self, template_id: str) -> List[str]:
        """Get sections for a template."""
        template = self.get_template(template_id)
        if template:
            return template.get("sections", [])
        return []
    
    def validate_template_id(self, template_id: str) -> bool:
        """Check if template ID is valid."""
        return template_id in self.templates


# Singleton instance
template_service = TemplateService()
